extension/oauth2clientauthextension/stsclientauth_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package oauth2clientauthextension
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"golang.org/x/oauth2"
+)
+
+type mockRoundTripper struct {
+	mock.Mock
+}
+
+func (m *mockRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	args := m.Called(req)
+	return args.Get(0).(*http.Response), args.Error(1)
+}
+
+func TestStsTokenSource(t *testing.T) {
+	// Create a temporary KSA token file
+	mockKsaTokenFile, err := os.CreateTemp("", "test_ksa_token_path")
+	assert.NoError(t, err)
+	defer os.Remove(mockKsaTokenFile.Name())
+	_, err = mockKsaTokenFile.WriteString("test-ksa-token")
+	assert.NoError(t, err)
+	err = mockKsaTokenFile.Close()
+	assert.NoError(t, err)
```

Comment:
```suggestion
	mockKsaTokenFile := filepath.Join(t.TempDir(), "test_ksa_token_path")
	err := os.WriteFile(mockKsaTokenFile, "test-ksa-token", 0644)
	assert.NoError(t, err)
```

If you use `T.TempDir()`, it will be removed when the test completes.